internal/testutils/kginkgo.go

```diff
@@ -162,9 +170,13 @@ func (k *KGinkgo) ExpectFinalizerToBeAdded(ctx context.Context, obj client.Objec
 
 	eo := makeKGinkgoOptions(opts...)
 	key := client.ObjectKeyFromObject(obj)
+	gvk := k.getGVKForObject(obj)
 
+	By(fmt.Sprintf("expecting finalizer %s to be added to %s/%s", finalizer, obj.GetNamespace(), obj.GetName()))
 	Eventually(func(g Gomega) {
-		fetched := &corev1.Service{}
```

Comment:
This was only being used in service tests and hence it worked. Because we want it to be useful across other tests, we need use unstructured.